### ISSUE TITLE
Compilation issues with Umpire@6.0.0

### ISSUE THREAD

**Issue by teonnik** [eth-cscs/DLA-Future#420]
As pointed out by @albestro :

DynamicPool & DynamicPoolMap seem to be deprecated in 6.0.0:
 https://github.com/LLNL/Umpire/pull/635
 
Changelog : https://github.com/LLNL/Umpire/blob/develop/CHANGELOG.md#changed-1

```
     150    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:4:
  >> 151    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/strategy/DynamicPool.hpp:16:1: error: expected unqual
            ified-id before 'using'
     152     using DynamicPool = DynamicPoolMap;
     153     ^~~~~
     154    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp: In function 'void dlaf::memory::internal::initializeUmpireHostAllocato
            r(std::size_t)':
  >> 155    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     156                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     157                                                  ^~~~~~~~~~~
     158    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: note: suggested alternative: 'DynamicPoolMap'
     159                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     160                                                  ^~~~~~~~~~~
     161                                                  DynamicPoolMap
  >> 162    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     163    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:46: note: suggested alternative: 'DynamicPoolMap'
     164                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     165                                                  ^~~~~~~~~~~
     166                                                  DynamicPoolMap
  >> 167    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:103: error: no matching function for call to 'umpire::ResourceManage
            r::makeAllocator<<expression error> >(const char [12], umpire::Allocator&, std::size_t&)'
     168                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     169                                                                                                           ^
     170    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     171    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note: candidate: 'templat
            e<class Strategy, bool introspection, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, Args&& ...)'
     172       Allocator makeAllocator(const std::string& name, Args&&... args);
     173                 ^~~~~~~~~~~~~
     174    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note:   template argument
             deduction/substitution failed:
  >> 175    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:103: error: template argument 1 is invalid
     176                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     177                                                                                                           ^
     178    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     179    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note: candidate: 'templat
            e<class Strategy, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, umpire::Tracking, Args&& ...)'
     180       Allocator makeAllocator(const std::string& name, Tracking tracked, Args&&... args);
     181                 ^~~~~~~~~~~~~
     182    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note:   template argument
             deduction/substitution failed:
  >> 183    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:37:103: error: template argument 1 is invalid
     184                 .makeAllocator<umpire::strategy::DynamicPool>("PINNED_pool", host_allocator, initial_bytes);
     185                                                                                                           ^
     186    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp: In function 'void dlaf::memory::internal::initializeUmpireDeviceAlloca
            tor(std::size_t)':
  >> 187    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     188                 .makeAllocator<umpire::strategy::DynamicPool>("DEVICE_pool", device_allocator,
     189                                                  ^~~~~~~~~~~
     190    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: note: suggested alternative: 'DynamicPoolMap'
     191                 .makeAllocator<umpire::strategy::DynamicPool>("DEVICE_pool", device_allocator,
     192                                                  ^~~~~~~~~~~
     193                                                  DynamicPoolMap
  >> 194    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: error: 'DynamicPool' is not a member of 'umpire::strategy'
     195    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:64:46: note: suggested alternative: 'DynamicPoolMap'
     196                 .makeAllocator<umpire::strategy::DynamicPool>("DEVICE_pool", device_allocator,
     197                                                  ^~~~~~~~~~~
     198                                                  DynamicPoolMap
  >> 199    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:65:72: error: no matching function for call to 'umpire::ResourceManager
            ::makeAllocator<<expression error> >(const char [12], umpire::Allocator&, std::size_t&)'
     200                                                               initial_bytes);
     201                                                                            ^
     202    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     203    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note: candidate: 'templat
            e<class Strategy, bool introspection, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, Args&& ...)'
     204       Allocator makeAllocator(const std::string& name, Args&&... args);
     205                 ^~~~~~~~~~~~~
     206    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:117:13: note:   template argument
             deduction/substitution failed:
  >> 207    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:65:72: error: template argument 1 is invalid
     208                                                               initial_bytes);
     209                                                                            ^
     210    In file included from /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:3:
     211    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note: candidate: 'templat
            e<class Strategy, class ... Args> umpire::Allocator umpire::ResourceManager::makeAllocator(const string&, umpire::Tracking, Args&& ...)'
     212       Allocator makeAllocator(const std::string& name, Tracking tracked, Args&&... args);
     213                 ^~~~~~~~~~~~~
     214    /m100/home/userexternal/tnikolov/code/spack/opt/spack/linux-rhel8-power9le/gcc-8.3.1/umpire-6.0.0-jhu5poqrifiukl6nladvccd47sj5dcbn/include/umpire/ResourceManager.hpp:120:13: note:   template argument
             deduction/substitution failed:
  >> 215    /tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-src/src/memory/memory_chunk.cpp:65:72: error: template argument 1 is invalid
     216                                                               initial_bytes);
     217                                                                            ^
  >> 218    make[2]: *** [src/CMakeFiles/dlaf.core.dir/build.make:233: src/CMakeFiles/dlaf.core.dir/memory/memory_chunk.cpp.o] Error 1
     219    make[2]: *** Waiting for unfinished jobs....
     220    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
     221    [ 75%] Built target dlaf.auxiliary
     222    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
  >> 223    make[1]: *** [CMakeFiles/Makefile2:1021: src/CMakeFiles/dlaf.core.dir/all] Error 2
     224    make[1]: *** Waiting for unfinished jobs....
     225    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
     226    [ 75%] Built target dlaf.solver
     227    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
     228    [ 75%] Built target dlaf.factorization
     229    make[2]: Leaving directory '/tmp/tnikolov/spack-stage/spack-stage-dla-future-develop-tuyamvofcu3ozaxq2azkars6b4gvofyz/spack-build-tuyamvo'
```

**Comment by davidbeckingsale** [comment]
The classes are deprecated but you should still be able to use them in Umpire 6.0.0. They used to end up transitively included, but that doesn't happen anymore. The class and alias is still available if you explicitly include the file, e.g.:
```
#include "umpire/strategy/DynamicPoolMap.hpp"
```

**Comment by albestro** [comment]
Thanks a lot for the indication @davidbeckingsale!

Since it is going to be deprecated, do you support the suggestion in the changelog?
> DynamicPoolMap marked deprecated. QuickPool should be used instead.

I don't have any knowledge about this currently, maybe @msimberg already knows something, otherwise it may be interesting to know from you what are the main differences (if there is any). I quickly checked the documentation (https://umpire.readthedocs.io/en/develop/sphinx/tutorial/dynamic_pool.html) and it seems it has not been yet updated.

Looking forward to your suggestions.

**Comment by msimberg** [comment]
@albestro, I came as far as you did, i.e. did not find any documentation on the differences. The simple replacement `DynamicPool` to `QuickPool` compiles fine (perhaps unsurprisingly), but I did not test to see if there was any performance difference.

@davidbeckingsale I suppose this https://github.com/LLNL/Umpire/blob/5f886b4299496b7cb6f9d62dc1372ce6d3832fbc/src/umpire/strategy/DynamicPool.hpp#L15-L16 should actually be
```
using DynamicPool UMPIRE_DEPRECATE_ALIAS("use QuickPool instead") = DynamicPoolMap;
```
for that deprecation to work correctly.
